### DISCUSSION TITLE
Suggestion: Include a "Concepts" section in the documentation

### DISCUSSION THREAD

**Discussion by jamesawebb** [blakeblackshear/frigate#2490] [Ideas]
My camera feeds are "blessed" with a multitude of moving vegetation which triggers the occasional object detection and a lot of false positives.  I'm trying to tweak my configuration to avoid this and finding myself referring back to the documentation more times than I should.  One thing I've noticed that's missing from the docs is a "Concepts" section which clearly defines constructs such as `detect`, `motion`, `objects`, `masks`, `zones` etc.  Some of these are clearly defined - others I've had to garner from the existing docs.  If we (read "you" :P) could make an effort to define these, it might help to ease configuration for newcomers.

Any thoughts?

**Comment by blakeblackshear** [comment]
Some of the sections didn't seem to warrant their own page. I tried to use the comments in the example config to document as much as possible and then supplemented with other pages. Can you be more specific about what you think is missing?

**Comment by jamesawebb** [comment]
Sorry, my responses are patchy due to holidays here.  Note that the task of trying to articulate my understanding of the various concepts has helped me "rubber duck" things a little.  I may still be wrong in places.

You've already done some excellent work explaining some of the nuances of motion and object detection in the [Masks : Further Clarification](https://docs.frigate.video/configuration/masks#further-clarification) section.  I'd pull that discussion out into it's own top-level article <wince>and perhaps throw a Concepts section at the top</wince>.  Suggested points (as I understand them):

- `detect`: Object detection.  Enabled by default.
- `object`: One of the pre-configured objects supported by the default detection model.
- `recordings`: 24x7 recordings.  
- `motion`: Motion detection (red boxes in debug mode). The first trigger for object detection.
- `event`: Captured (recorded?) object detection.
- `snapshot`: A sample image of the above `event`.  Disabled as `recordings` are preferred.
- `clip`: A sample "video clip" of the above `event`.
- `masks`
  * `motion masks`: used to mask guaranteed false positives.  Detection does not occur here.
  * `object filter masks`: used to mask guaranteed false positives *for a given object*.
- `zones`: Areas of note.

A list like this might act as a good reference for newcomers to refer to along their configuration journey.

Lastly, as a plug for the excellent comments you have in the sample configuration file - encourage readers to go there for further enlightenment.

Lastly-for-realz, this is only a suggestion.  Please treat it as such.
